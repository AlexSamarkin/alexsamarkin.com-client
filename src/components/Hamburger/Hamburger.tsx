import React from 'react';

export interface HamburgerProps {
    onClick: () => void;
    isOpen: boolean;
}

export const Hamburger: React.FC<HamburgerProps> = ({ onClick, isOpen }) => {
    const classNames = `hamburger ${isOpen ? 'is-active' : ''}`;
    return (
        <div className="circle-menu">
            <div className={classNames} onClick={onClick}>
                <div className="line" />
                <div className="line" />
                <div className="line" />
            </div>
        </div>
    );
};

export default Hamburger;
