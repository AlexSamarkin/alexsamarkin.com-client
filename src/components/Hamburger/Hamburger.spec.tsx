import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Hamburger from './Hamburger';

Enzyme.configure({ adapter: new Adapter() });

describe('Hamburger component', () => {
    test('renders', () => {
        const wrapper = shallow(<Hamburger isOpen={false} onClick={() => true} />);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
});
