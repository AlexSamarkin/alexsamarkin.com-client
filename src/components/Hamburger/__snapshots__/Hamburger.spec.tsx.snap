// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger component renders 1`] = `ShallowWrapper {}`;

exports[`Hamburger component renders open 1`] = `ShallowWrapper {}`;
